### PR TITLE
Onboard flow endpoint

### PR DESCRIPTION
**Pull request by cecilynie**
Additional onboard endpoint for function replay and init autotune

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java [hunk 1]
```diff
@@ -396,8 +395,8 @@ public Response checkAnomaliesHasLabel(@PathParam("functionId") long id,
   /**
    * End point to trigger initiate alert filter auto tune
    * @param id functionId to initiate alert filter auto tune
-   * @param startTime: training data starts time
-   * @param endTime: training data ends time
+   * @param startTimeIso: training data starts time
```
Comment (bl44) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java]: Can you comment here what is the right format ofr Iso time?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java [hunk 2]
```diff
@@ -407,11 +406,20 @@ public Response checkAnomaliesHasLabel(@PathParam("functionId") long id,
...
       @QueryParam("autoTuneType") @DefaultValue("AUTOTUNE") String autoTuneType, @QueryParam("nExpected") int nExpectedAnomalies,
       @QueryParam("holidayStarts") @DefaultValue("") String holidayStarts,
       @QueryParam("holidayEnds") @DefaultValue("") String holidayEnds){
+    long startTime;
+    long endTime;
+    try {
+      startTime = ISODateTimeFormat.dateTimeParser().parseDateTime(startTimeIso).getMillis();
+      endTime = ISODateTimeFormat.dateTimeParser().parseDateTime(endTimeIso).getMillis();
+    } catch (Exception e) {
+      throw new WebApplicationException("Unable to parse strings, " + startTimeIso + " and " + endTimeIso +
```
Comment (bl44) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java]: Same here. Better to be verbose and user can correct their input in a informed way.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java [hunk 3]
```diff
@@ -540,7 +549,7 @@ public Response evaluateAlertFilterByAutoTuneId(@PathParam("autotuneId") long id
   public Response anomalyFunctionReplay(@PathParam("id") @NotNull long functionId,
       @QueryParam("start") @NotNull String replayStartTimeIso, @QueryParam("end") @NotNull String replayEndTimeIso,
       @QueryParam("speedup") @DefaultValue("true") boolean speedup,
-      @QueryParam("tune") @DefaultValue("{\"pValueThreshold\":[0.1,0.05,0.01,0.05,0.01]}") String tuningJSON,
```
Comment (bl44) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java]: Why these particular combination of pValueThreshold? try to follow a log-scale for 1, 2, 3 sigma? or should we restrict the default to few cases: like 0.05, 0.01 (2,3 sigma)

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/lib/FunctionReplayRunnable.java [hunk 4]
```diff
@@ -37,6 +37,7 @@
   private Long functionAutotuneConfigId;
   private boolean speedUp;
   private boolean selfKill;
+  private long clonedFunctionId;
```
Comment (bl44) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/lib/FunctionReplayRunnable.java]: is these runnable going to create multiple cloneFunctionIds, if it takes multiple configurations? So you will always return the last one, is this the expected behavior?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/utils/AnomalyUtils.java [hunk 5]
```diff
@@ -66,4 +66,22 @@ public static Boolean checkHasLabels(List<MergedAnomalyResultDTO> mergedAnomalyR
     }
     return false;
```
Comment (bl44) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/utils/AnomalyUtils.java]: Does the AnomalyFeedbackType contains 'NULL', what's the convention we used here?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/utils/AnomalyUtils.java [hunk 6]
```diff
@@ -66,4 +66,22 @@ public static Boolean checkHasLabels(List<MergedAnomalyResultDTO> mergedAnomalyR
...
   }
+
```
Comment (bl44) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/utils/AnomalyUtils.java]: Where and how this metaDataNode will be used? Does it have broader use cases? You may give it a more intuitive name. Also, even thought this is sub class, its better to name it 'MetaDataNode' to reflect that this is a class name, not a variable name.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/lib/FunctionReplayRunnable.java (hunk 4 above)
Comment (cecilynie) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/lib/FunctionReplayRunnable.java]: The only usage here is to do replay once using the input configuration. Hence only one clone function Id will be generated. Modify its name to lastClonedFunctionId.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/utils/AnomalyUtils.java (hunk 5 above)
Comment (cecilynie) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/utils/AnomalyUtils.java]: AnomalyFeedbackType has been overrided.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java [hunk 7]
```diff
@@ -314,8 +313,8 @@ public Response generateAnomaliesInTrainingData(@PathParam("id") @NotNull long i
    * @param functionId: the functionId to fetch merged anomalies with holidays removed
    * @param startTime: start time in milliseconds of merged anomalies
    * @param endTime: end time of in milliseconds merged anomalies
-   * @param holidayStarts: holidayStarts in milliseconds as string, in format {start1,start2,...}
-   * @param holidayEnds: holidayEnds in milliseconds as string, in format {end1, end2,...}
+   * @param holidayStarts: holidayStarts in ISO Format {start1,start2,...}, ex: 2016-5-23T00:00:00Z
```
Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java]: Remove brackets in the example, including all the examples in the other code.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java [hunk 8]
```diff
@@ -314,8 +313,8 @@ public Response generateAnomaliesInTrainingData(@PathParam("id") @NotNull long i
    * @param functionId: the functionId to fetch merged anomalies with holidays removed
    * @param startTime: start time in milliseconds of merged anomalies
    * @param endTime: end time of in milliseconds merged anomalies
-   * @param holidayStarts: holidayStarts in milliseconds as string, in format {start1,start2,...}
-   * @param holidayEnds: holidayEnds in milliseconds as string, in format {end1, end2,...}
+   * @param holidayStarts: holidayStarts in ISO Format ex: 2016-5-23T00:00:00Z, start1,start2,...
```
Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java]: I think it is better to say, a list of ISO DateTime String (e.g. 2016-5-23T00:00:00Z) separated by comma.

Comment (cecilynie) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionJobResource.java]: fixed

**Pull request merged**